monix-eval/shared/src/test/scala/monix/eval/TaskConversionsSuite.scala

```diff
@@ -378,6 +379,45 @@ object TaskConversionsSuite extends BaseTestSuite {
     assertEquals(f2.value, Some(Success(99)))
   }
 
+  test("Task.fromPublisher protects against user error") { implicit s =>
```

Comment:
Since we have `task.toReactivePublisher`, it's probably worth adding a property-based test about how they relate, e.g.:
```scala
check1 { task: Task[Thing] =>
  Task.fromPublisher(task.toReactivePublisher) <-> task.map(Some(_))
}
```